src/Microsoft.IdentityModel.Logging/TextWriterEventListener.cs

```diff
@@ -0,0 +1,71 @@
+﻿//-----------------------------------------------------------------------
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved
+// Apache License 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------
+
+using System.Diagnostics.Tracing;
+using System.IO;
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// Event listener that writes logs to a file or a fileStream provided by user.
+    /// </summary>
+    public class TextWriterEventListener : EventListener
+    {
+        private static string _defaultLogFileName = "IdentityModelLogs.txt";
+        private StreamWriter _streamWriter;
+        private Stream _fileStream;
```

Comment:
I don't think we need  _fileStream or _filePath variables. Use them in the constructor to build the StreamWriter.